tests/test_pipeline.py

```diff
@@ -0,0 +1,97 @@
+import os
+import pytest
+import pandas as pd
+import numpy as np
+from skimage.io import imsave
+import luigi
+from cyto_ml.pipeline.pipeline_decollage import ReadMetadata, DecollageImages, UploadDecollagedImagesToS3
+
+
+@pytest.fixture
+def temp_dir(tmp_path):
+    """Creates a temporary directory using pytest's tmp_path fixture."""
+    return tmp_path
+
+
+def test_read_metadata(temp_dir):
+    # Create a mock .lst file for testing
+    lst_file_content = "001\nnum-fields|value\n"
```

Comment:
So the issue here was intricacies of the "csv-like" file format that the FlowCam instrument exports with its data. It has a meaningless numeric header, and 53 lines of description of header datatypes before the actual, CSV-readable data kicks in. The parsing function assumes this and fails if it's not the case. The "fix" here is to flesh out the test fixture until it matches the format expectations

I would tend _not_ to try and construct a test fixture in code here (I can see why you would for a simpler data structure) but use sample data inside `tests/fixtures/` for any format with a bit of complexity. There is a cut-down `.lst` file lying around in the test fixtures for this project.

Perhaps there are good reasons not to! E.g. to deliberately break the fixtures in case the instrument outputs bad metadata, and see how our code handles that. I know my test style tends to be "happy path" ...
I've left it as-is though, just padded it out with dummy data until it matches what the microscope generates